sql/core/src/test/scala/org/apache/spark/sql/execution/command/DDLParserSuite.scala

```diff
@@ -789,7 +789,7 @@ class DDLParserSuite extends AnalysisTest with SharedSparkSession {
     assertError("select interval '23:61:15' hour to second",
       "minute 61 outside range [0, 59]")
     assertError("select interval '.1111111111' second",
-      "nanosecond 1111111111 outside range")
+      "Invalid interval string")
```

Comment:
This actually exposes a problem of `stringToInterval`: the error reporting becomes worse.

I think the reason is `stringToInterval` returns null to indicate invalid input, but then there is no chance to know what is the exact failure. Can we make `stringToInterval` throw exception? cc @MaxGekk

Reply:
it used to replace `safeFromString`, we may could wrap a try catch here and use a `safe` flag to control throw exception or return null

Reply:
instead of a safe flag, I'd like to have 2 methods like before
1. `stringToInterval` which throws exception on failure
2. `safeStringToInterval` which calls `stringToInterval` and try-catch to return null on failure

Reply:
ok, i will check on this right now.

Reply:
This makes sense. Initially I thought about raising exceptions from `stringToInterval` and catch them on the next level to convert to null. For now, SGTM.

Reply:
For this part, I encounter another bug here, current string to interval cast logic does not support i.e. `cast('.111' as interval)` which will fail in SIGN state and return `null`, actually, it is `00:00:00.111`. I had fixed in the current pr, I also wonder if I do it separately to fix the bug first.

Reply:
These are the results of the master branch. 
```sql
-- !query 63
select interval '.111 seconds'
-- !query 63 schema
struct<0.111 seconds:interval>
-- !query 63 output
0.111 seconds


-- !query 64
select cast('.111 seconds' as interval)
-- !query 64 schema
struct<CAST(.111 seconds AS INTERVAL):interval>
-- !query 64 output
NULL
```

Reply:
Seems like the newly added interval parser has a bug...